plugin/ochttp/stats_transport.go

```diff
@@ -126,3 +126,75 @@ func (t *tracker) Close() error {
 	t.end()
 	return t.body.Close()
 }
+
+func (h *Handler) wrapForStats(w http.ResponseWriter, r *http.Request) (http.ResponseWriter, *http.Request, ender) {
+	ctx, _ := tag.New(r.Context(),
+		tag.Upsert(Host, r.URL.Host),
+		tag.Upsert(Path, r.URL.Path),
+		tag.Upsert(Method, r.Method))
+	track := &trackingResponseWriter{
+		start:  time.Now(),
+		ctx:    ctx,
+		writer: w,
+	}
+	if r.Body == nil {
+		// TODO: Handle cases where ContentLength is not set.
+		track.reqSize = -1
+	} else if r.ContentLength > 0 {
+		track.reqSize = r.ContentLength
+	}
+	stats.Record(ctx, ServerRequestCount.M(1))
+	return track, r.WithContext(ctx), track
+}
+
+type trackingResponseWriter struct {
+	reqSize    int64
+	respSize   int64
+	ctx        context.Context
+	start      time.Time
+	statusCode string
+	endOnce    sync.Once
+	writer     http.ResponseWriter
+}
+
+var _ http.ResponseWriter = (*trackingResponseWriter)(nil)
+
+func (t *trackingResponseWriter) End() {
+	t.endOnce.Do(func() {
+		m := []stats.Measurement{
+			ServerLatency.M(float64(time.Since(t.start)) / float64(time.Millisecond)),
+			ServerResponseBytes.M(t.respSize),
+		}
+		if t.reqSize >= 0 {
+			m = append(m, ServerRequestBytes.M(t.reqSize))
+		}
+		ctx, _ := tag.New(t.ctx, tag.Upsert(StatusCode, t.statusCode))
+		stats.Record(ctx, m...)
+	})
+}
+
+func (t *trackingResponseWriter) Header() http.Header {
+	return t.writer.Header()
+}
+
+func (t *trackingResponseWriter) Write(data []byte) (int, error) {
+	if t.statusCode == "" {
```

Comment:
It was a string in the tracker, so I just copied that. It needs to be a string eventually because it's inserted into a tag. No reason not to do the conversion up-front.